src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsIconFactory.java

```diff
@@ -905,17 +914,30 @@ public void paintIcon(Component c, Graphics g, int x, int y) {
                         if (xp != null) {
                             Skin skin;
                             skin =  xp.getSkin(c, backgroundPart);
-                            skin.paintSkin(g, x, y,
-                                getIconWidth(), getIconHeight(), backgroundState);
-                            if (icon == null) {
+                            if (!isWindows11OrLater) {
+                                skin.paintSkin(g, x, y,
+                                        getIconWidth(), getIconHeight(), backgroundState);
+                                if (icon == null) {
+                                    skin = xp.getSkin(c, part);
+                                    skin.paintSkin(g, x + OFFSET, y + OFFSET, state);
+                                }
+                            } else {
+                                skin.paintSkin(g, x, y,
+                                        getIconWidth(), getIconHeight(), backgroundState);
+                                skinWidth = getIconWidth();
                                 skin = xp.getSkin(c, part);
```

Comment:
What I expect to see here is that the skin is always painted at (`x + OFFSET, y + OFFSET`)

```java
skin.paintSkin(g, x + OFFSET, y + OFFSET, state);
```

in both Windows 10 and 11.

The check mark and bullet have to remain on their own designated place.